### PR TITLE
Add Embed Mode Share Feature

### LINKED ISSUE #964: Embedding / Worldview widget
From WV-1651: Allow users to embed worldview on their own site

### PR DESCRIPTION
**Pull request by edplato**
## Description

Fixes #964  .

- [x] Add the Embed Mode Feature to allow users to embed Worldview into their webpage and allow their visitors to interact with the Worldview instance they are sharing. 
- [x] Add dynamic A/B dates to Comparison Mode

### Embed Mode:

- Embed share tab will create an `iframe` code block with embedded Worldview that can be copied using copy to clipboard,. A link to additional Embed documentation is provided for further guidance.
- Split the Social Toolbar features and rename as Share Toolbar with Link, Embed, and Social tabs.
- Certain features are disabled, hidden, or removed to focus the embedded Worldview instance to what is being shared. Users can further explore Worldview using the Worldview sidebar logo and the added upper right new tab button, which will both open up a new instance of full production Worldview in a new tab with the same query string parameters used in the embedded instance.
- Embed mode includes an overlay that requires a click to interact in order to prevent unexpected interaction when scrolling down a page using an embedded Worldview page.
- Remove/hide/restyle UI feature styles primarily handled in `embed.css` and `mobile.css` where possible/not conditionally determined within a component.
- Note: ` web/js/containers/sidebar/events.js` is currently filtering events out to 1) only show Events tab if an event is selected and 2) only show selected Event and remove unselect interaction.

### Comparison Mode:

- A/B dates, if different, are now displayed within the individual compare mode UI elements to help users determine what data they’re viewing. If the dates are the same, they are displayed as the currently displayed A/B titles. 
- Style changes have been made to increase unobstructed map area and accommodate date text.

### Additional:

- Create reusable `HoverTooltip` util component
- When location search input was focused and animation widget was open the animation widget could be activated using hotkeys - most common is `SPACE`. A fix was added to prevent animation widget hotkeys from firing if location search input is focused.
- Revise date text to use `YYYY MMM DD` format for Compare Mode (sidebar and UI modes), Mobile Date Picker, Timeline Axis Date Tooltip, and Timeline Coverage Panel.
- Random spell fixes, updated E2E tests, and refactors.

## How To Test

Since there is a lot of test deployment linking going on, I will copy test links into our `worldview-dev` Slack channel. Feel free to ask if they get lost in the channel.

Using our main UAT, review the Share Toolbar which includes Embed Mode link copying, compare mode date display, and date text `YYYY MMM DD` format changes.

To review an embedded iframe instance of Worldview, the `worldview-embed-test` UAT test instance has a great variety of embedded use cases. You can also add `&em=true` to a permalink within our main UAT and it will allow you to interact with the style and features of an embedded Worldview, albeit without the iframe.

I am requesting overall code, interaction, and style reviews using different browsers, viewport sizes. Please include browser if it seems to be a browser specific issue.

## PR Submission Checklist

This is simply a reminder of what we are going to look for before merging your code.

- I have read the [CONTRIBUTING](https://github.com/nasa-gibs/worldview/blob/main/.github/CONTRIBUTING.md) doc
- I have added necessary documentation (if applicable)
- I have added tests that prove my fix is effective or that my feature works (if applicable)
- Any dependent changes have been merged and published in downstream modules (if applicable)

@nasa-gibs/worldview

### CODE REVIEW

**Review by minniewong — APPROVED** [review]
Looks nice and works well! 🚀

**Review by jasontk19 — CHANGES_REQUESTED** [review]
💯  for a very thorough PR description... it really helps with a diff this large!

Overall this is looking great.  Really like the one-time-format-to-rule-them-all changes and the simplified `HoverTooltip` component.  I just had a few minor suggestions in the code and other observations are listed below:

---

* I don't know if this has changed from `develop` but this modal seems placed a little too far left.  The rest of the toolbar modals open directly underneath their corresponding buttons:
Screen Shot 2021-06-11 at 8 29 02 AM


* It looks like re-ordering layers is disabled but re-ordering groups is not.  I don't know that we _need_ to disable re-ordering groups but just an observation.

* It looks like the GIF creation button is disabled in the animation widget but I didn't see any code handling this in that component so not sure if that's intentional or not?  Seems like making a GIF would be OK in embed mode but might be worth discussion.

File: doc/embed.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Embedding Worldview
+
+To embed NASA Worldview in your page, navigate to the share toolbar, select the `embed` tab, and copy the code that wraps the current Worldview instance in an `<iframe>` inline element. Add this HTML code to your website to embed Worldview. You will need to wrap the `<iframe>` code in a wrapper element (e.g., `<div>`) with your desired `width`, `height`, and any additional style changes.
```
Comment (rboller) [inline doc/embed.md:3]: Users / embedders may not know what the "share toolbar" is so would suggest rewording to something like, "...click the 'share' button in the upper right corner of the app, select the 'embed' tab..."

File: web/js/containers/embed.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    googleTagManager.pushEvent({
+      event: 'embed_open_new_tab',
+    });
+    window.open(permalink, '_blank');
+  }
+
+  renderEmbedLinkBtn() {
+    const { isMobile } = this.props;
+    const buttonId = 'wv-embed-link-button';
+    const labelText = 'Open new tab with content in Worldview';
```
Comment (rboller) [inline web/js/containers/embed.js:40]: This tool tip could use some clarification... something like, "Open this Worldview map in a new window".

**Review by rboller — APPROVED** [review]
Looking great, Ed!    

I've made a few minor comments on different parts of the code and have this one about dates in comparison mode:  if different dates are selected for the A/B opacity slider, the text formatting could be spruced up to provide more visual separation between the "A:" and "B:" labels from their respective dates... e.g.:

**A:**  2021 May 16   ----*--   **B:** 2021 May 19

(though the above doesn't do a whole lot for visual separation with github formatting)

File: web/js/containers/sidebar/layer-list.js [hunk 3]
```diff
@@ -220,15 +221,23 @@ LayerList.propTypes = {
...
+  const {
+    embed, proj, config, map,
+  } = state;
+  const { isEmbedModeActive } = embed;
   const zots = lodashGet(map, 'ui.selected')
     ? getZotsForActiveLayers(state)
     : {};
+  let activeLayers = getActiveLayers(state);
+  if (isEmbedModeActive) {
+    activeLayers = activeLayers.filter((layer) => layer.visible && layer.layergroup !== 'Reference');
```
[the remark this answers is not in the document]
Comment (edplato) [reply web/js/containers/sidebar/layer-list.js:234]: For this case, the reference layer filtering is necessary at the level of the layer-list, but not throughout the entire app.

**Review by jasontk19 — APPROVED** [review]
Looks good!
